File: OdontoFlowAPI/OdontFlow/OdontFlow.Domain/ViewModel/PriceList/PriceListItemViewModel.cs
using OdontFlow.Domain.ViewModel.Product;

namespace OdontFlow.Domain.ViewModel.PriceList;

public class PriceListItemViewModel
{
    public Guid Id { get; set; }
    public Guid ProductId { get; set; } 
    public Guid PriceListId { get; set; }
    public string PriceListName { get; set; } = default!;
    public decimal Price { get; set; }
    public ProductViewModel Product { get; set; } = default!;
    public string? Comments { get; set; }
}

File: OdontoFlowWeb/OdontFlow/src/app/core/model/pricelistItem-view.model.ts
import { ProductViewModel } from "./product-view.model";

export interface PriceListItemViewModel {
  id?: string;
  productId?: string;
  productName?: string;
  priceListId?: string;
  priceListName?: string;
  price?: number;
  product?: ProductViewModel, 
  comments?: string
}

export interface CreatePriceListItemDTO {
  productId: string;
  priceListId: string;
  price: number;
  comments?: string;
}


export interface UpdatePriceListItemDTO {
  id: any;
  price: number;
  comments?: string;
}